packages/lib/src/item-input/select.ts

```diff
@@ -0,0 +1,83 @@
+import inquirer, { ChoiceCollection, Separator } from 'inquirer'
+
+import { CancelAction, cancelOption, helpAction, helpOption, InputDefinition, inquirerPageSize, maxItemValueLength } from './defs'
+import { clipToMaximum, stringFromUnknown } from '../util'
+
+
+export type SelectDefOptions<T> = {
+	/**
+	 * A summary of the chosen item to display to the user in a list.
+	 *
+	 * If the choices are simple strings or numbers, the default is to display the string or
+	 * number. For complex choices, the default function uses the `name` field.
+	 */
+	summarizeForEdit?: (item: T, context?: unknown[]) => string
+
+	default?: T
+
+	helpText?: string
+}
+
+export type SelectDefChoice<T> =
+	| string
+	| number
+	| { name: string; value: T }
+
+/**
+ * InputDefinition choosing a single item from a list.
```

Comment:
I think it would scan better as "InputDefinition for choosing a single item from a list."

Reply:
Yeah, that was bad. Fixed.